### PR TITLE
Test varying inputs and outputs for wgsl

### PR DESCRIPTION
**Pull request by expipiplus1**
Closes https://github.com/shader-slang/slang/issues/5067

New tests, covering what's declared supported in the WGSL support docs

- tests/wgsl/semantic-coverage.slang
- tests/wgsl/semantic-depth.slang
- tests/wgsl/semantic-dispatch-thread-id.slang
- tests/wgsl/semantic-group-id.slang
- tests/wgsl/semantic-group-index.slang
- tests/wgsl/semantic-group-thread-id.slang
- tests/wgsl/semantic-instance-id.slang
- tests/wgsl/semantic-is-front-face.slang
- tests/wgsl/semantic-position.slang
- tests/wgsl/semantic-sample-index.slang
- tests/wgsl/semantic-vertex-id.slang

WGSL enabled existing tests:
- tests/compute/compile-time-loop.slang
- tests/compute/constexpr.slang
- tests/compute/discard-stmt.slang
- tests/metal/nested-struct-fragment-input.slang
- tests/metal/nested-struct-fragment-output.slang
- tests/metal/nested-struct-multi-entry-point-vertex.slang
- tests/metal/no-struct-vertex-output.slang
- tests/metal/sv_target-complex-1.slang
- tests/metal/sv_target-complex-2.slang
- tests/bugs/texture2d-gather.hlsl
- tests/render/cross-compile-entry-point.slang
- tests/render/nointerpolation.hlsl
- tests/render/render0.hlsl
- tests/render/cross-compile0.hlsl
- tests/render/imported-parameters.hlsl
- tests/render/unused-discard.hlsl

Can't be enabled due to missing wgsl features

- tests/compute/texture-sampling.slang

### CODE REVIEW

**Comment by aleino-nv** [comment]
Is this for https://github.com/shader-slang/slang/issues/5067?

**Comment by aleino-nv** [comment]
Failing test:

> tests/metal/atomic-intrinsics.slang(23): error 36107: entrypoint 'computeMain' does not support compilation target 'wgsl' with stage 'compute'

I think we need to disable this one, or create a WGSL-specific version. WGSL doesn't support the Interocked* functions.

File: tests/render/nointerpolation.hlsl [hunk 1]
```diff
@@ -1,6 +1,6 @@
 //TEST(smoke):COMPARE_HLSL_RENDER:
 // WGSL: nointerpolate doesn't work #5625
-//DISABLE_TEST(smoke):COMPARE_HLSL_RENDER: -wgpu
+//TEST(smoke):COMPARE_HLSL_RENDER: -wgpu
```
Comment (aleino-nv) [inline tests/render/nointerpolation.hlsl:3]: https://github.com/shader-slang/slang/pull/5641 will enable this. I don't think it should work yet.

Comment (aleino-nv) [reply tests/render/nointerpolation.hlsl:3]: What kind of output do you see? It should be a solid red triangle, I think.

File: tests/compute/discard-stmt.slang [hunk 2]
```diff
@@ -1,5 +1,6 @@
 //TEST(compute):COMPARE_RENDER_COMPUTE: -shaderobj
 //TEST(compute):COMPARE_RENDER_COMPUTE: -mtl -shaderobj
+//TEST(compute):COMPARE_RENDER_COMPUTE: -wgsl -shaderobj
```
Comment (aleino-nv) [inline tests/compute/discard-stmt.slang:3]: I'm pretty sure this test is already running for WGSL due to line 1. Did it not run locally, for you?

Comment (expipiplus1) [reply tests/compute/discard-stmt.slang]: ah, oversight on my part, thanks

**Review by csyonghe — APPROVED** [review]

**Pull request merged**